common/src/model_registration/register.py

```diff
@@ -15,11 +15,45 @@
 from azure.identity import ManagedIdentityCredential
 from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
 from azureml.core import Run
+from applicationinsights import TelemetryClient
 
 
 SUPPORTED_MODEL_ASSET_TYPES = [AssetTypes.CUSTOM_MODEL, AssetTypes.MLFLOW_MODEL]
 MLFLOW_MODEL_FOLDER = "mlflow_model_folder"
 
+tc = None
+
+
+def init_tc():
+    """Initialize telemetry client."""
+    global tc
+    if tc is None:
+        try:
+            tc = TelemetryClient("71b954a8-6b7d-43f5-986c-3d3a6605d803")
+        except Exception as e:
+            print(f"Exception while initializing app insights: {e}")
+
+
+def tc_log(message):
+    """Log message to app insights."""
+    global tc
+    try:
+        tc.track_event(name="FM_import_pipeline_debug_logs", properties={"message": message})
+        print(message)
```

Comment:
Consider putting this in a `finally` or printing before you call TelemetryClient.

Reply:
Done